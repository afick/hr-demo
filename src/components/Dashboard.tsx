
import React from 'react'

const Dashboard = (_props: any) => {


    return (
        <div style={{
            width: '100%', 'textAlign': 'center', marginTop: '10px'
        }}>
            <h1>Welcome to the Fickersons</h1>
            <p>This site was build with React.JS <em>(yarn create react-app my-app)</em>with a Amplify backend. <em>(amplify push)</em></p>
            <p>GraphQL with DynamoDB source</p>
            <p>Ant Design and AG-Grid</p>
            <p>Pardon the design</p>
        </div >

    )

}

export default Dashboard
