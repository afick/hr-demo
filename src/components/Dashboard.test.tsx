import { render, screen } from '@testing-library/react'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import Dashboard from './Dashboard'
import renderer from 'react-test-renderer'

describe('Dashboard tests', () => {
    it('renders without crashing', () => {
        <BrowserRouter>
            const div = document.createElement('div')
            ReactDOM.render(<Dashboard />, div)
            ReactDOM.unmountComponentAtNode(div)
        </BrowserRouter>
    })
    it('render screen test', () => {
        render(<Dashboard />)
        expect(screen.getByText('Welcome to the Fickersons')).toBeInTheDocument()
    })

    it('renders snapshot as expected', () => {
        const tree = renderer.create(<Dashboard />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
