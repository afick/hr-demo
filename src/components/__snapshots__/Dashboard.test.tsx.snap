// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests renders snapshot as expected 1`] = `
<div
  style={
    Object {
      "marginTop": "10px",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <h1>
    Welcome to the Fickersons
  </h1>
  <p>
    This site was build with React.JS 
    <em>
      (yarn create react-app my-app)
    </em>
    with a Amplify backend. 
    <em>
      (amplify push)
    </em>
  </p>
  <p>
    GraphQL with DynamoDB source
  </p>
  <p>
    Ant Design and AG-Grid
  </p>
  <p>
    Pardon the design
  </p>
</div>
`;
